sagemaker_processing/spark_distributed_data_processing/sagemaker-spark-processing.ipynb

```diff
@@ -0,0 +1,572 @@
+    "input_preprocessed_prefix_abalone = prefix + '/input/preprocessed/abalone'\n",
+    "\n",
+    "sagemaker_session.upload_data(path='./data/abalone.csv', bucket=bucket, key_prefix=input_prefix_abalone)\n",
+    "\n",
+    "# Run the processing job\n",
+    "spark_processor = PySparkProcessor(\n",
+    "    base_job_name=\"sm-spark\",\n",
+    "    framework_version=\"2.4\",\n",
+    "    role=role,\n",
+    "    instance_count=20,\n",
+    "    instance_type=\"ml.m5.xlarge\",\n",
+    "    max_runtime_in_seconds=1200,\n",
+    ")\n",
+    "\n",
+    "spark_processor.run(\n",
+    "    submit_app=\"./code/preprocess.py\",\n",
+    "    arguments=['s3_input_bucket', bucket,\n",
+    "               's3_input_key_prefix', input_prefix_abalone,\n",
+    "               's3_output_bucket', bucket,\n",
+    "               's3_output_key_prefix', input_preprocessed_prefix_abalone],\n",
+    "    spark_event_logs_s3_uri='s3://' + bucket + '/' + prefix + '/spark_event_logs',\n",
+    "    logs=False\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Validate Data Processing Results\n",
+    "\n",
+    "Next, validate the output of our data preprocessing job by looking at the first 5 rows of the output dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print('Top 5 rows from s3://{}/{}/train/'.format(bucket, input_preprocessed_prefix_abalone))\n",
+    "!aws s3 cp --quiet s3://$bucket/$input_preprocessed_prefix_abalone/train/part-00000 - | head -n5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### View the Spark UI\n",
+    "\n",
+    "Next, you can view the Spark UI by running the history server locally in this notebook. (**Note:** this feature will only work in a local development environment with docker installed or on a Sagemaker Notebook Instance. This feature does not currently work in SageMaker Studio Notebooks.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark_processor.start_history_server()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After viewing the Spark UI, you can terminate the history server before proceeding."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark_processor.terminate_history_server()     "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Example 2: Specify additional python and jar file dependencies\n",
+    "\n",
+    "The next example demonstrates a scenario where additional python file dependencies are required by the PySpark script. You'll use a sample PySpark script that requires additional UDFs defined in a local module."
```

Comment:
"UDFs" -> "user-defined functions (UDFs)" (to explain the acronym)

Reply:
Updated